tests/fsharp/Compiler/Language/HatDesugaringTests.fs

```diff
@@ -0,0 +1,107 @@
+﻿namespace FSharp.Compiler.UnitTests
+
+open NUnit.Framework
+open FSharp.Compiler.SourceCodeServices
+
+[<TestFixture>]
+module HatDesugaringTests =
```

Comment:
Tests are great. I would add one more test of trying to use `@` outside the indexer as a prefix op, simply:
```fsharp
module X

let x = @1
```

Just to ensure that this does not compile with the specific error.